### PR TITLE
new packages: py-rst2pdf and py-smartypants

### PR DESCRIPTION
**Pull request by glennpj**
The py-smartypants package is a dependency of py-rst2pdf.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-rst2pdf/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    LaTeX. This tool provides an alternative by producing PDF directly using
+    the ReportLab library."""
+
+    homepage = "https://rst2pdf.org/"
+    pypi     = "rst2pdf/rst2pdf-0.99.tar.gz"
+
+    version('0.99', sha256='8fa23fa93bddd1f52d058ceaeab6582c145546d80f2f8a95974f3703bd6c8152')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rst2pdf/package.py:22]: Missing build dep on `py-setuptools-scm`

File: var/spack/repos/builtin/packages/py-smartypants/package.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class PySmartypants(PythonPackage):
+    """smartypants is a Python fork of SmartyPants."""
+
+    homepage = "https://github.com/leohemsted/smartypants.py"
+
+    # PyPI only has the wheel
+    url      = "https://github.com/leohemsted/smartypants.py/archive/refs/tags/v2.0.1.tar.gz"
+
+    version('2.0.1', sha256='b98191911ff3b4144ef8ad53e776a2d0ad24bd508a905c6ce523597c40022773')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-smartypants/package.py:17]: Missing a build dep on setuptools

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-rst2pdf/package.py (hunk 1 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-rst2pdf/package.py:22]: Ah, is that coming from this in `setup.py`:
```
    use_scm_version=True,
```
I wondered about that.

File: var/spack/repos/builtin/packages/py-smartypants/package.py (hunk 2 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-smartypants/package.py:17]: I did not see setuptools in the setup.py file. Hunting around I see this in the spack docs:

    If there is no pyproject.toml file and only a setup.py or setup.cfg file, you can assume that the project uses setuptools. 

I will have to remember that.

File: var/spack/repos/builtin/packages/py-rst2pdf/package.py (hunk 1 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-rst2pdf/package.py:22]: I see it in `pyproject.toml`, which I did not look at until just now.

File: var/spack/repos/builtin/packages/py-smartypants/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-smartypants/package.py:17]: The specific reason why setuptools is needed is documented here: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#distutils

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**